### ISSUE TITLE
isinstance does not support generic types.

### ISSUE THREAD

**Issue by domdfcoding** [RustPython/RustPython#2936]
## Feature

CPython supports checking whether an object is an instance of one of the abstract types from the typing module, such as `Dict` or `Sequence`:

```python
>>> from typing import Sequence
>>> isinstance([], Sequence)
True
```

These "classes" are actually instances of `typing._GenericAlias` in Python 3.8 (`typing._SpecialGenericAlias` in 3.9).

In RustPython an exception is raised:

```python
>>>>> from typing import Sequence
>>>>> isinstance([], Sequence)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: isinstance() arg 2 must be a type or tuple of types, not _GenericAlias

```

-----

In CPython instance checks with subscripted generics fail:

```python

>>> isinstance([], Sequence[str])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.8/typing.py", line 769, in __instancecheck__
    return self.__subclasscheck__(type(obj))
  File "/usr/lib/python3.8/typing.py", line 777, in __subclasscheck__
    raise TypeError("Subscripted generics cannot be used with"
TypeError: Subscripted generics cannot be used with class and instance checks

```

-----

In CPython, `isinstance` will accept anything as the second argument as long as it has an ``__instancecheck__`` method:

```python
>>> class G:
...     def __instancecheck__(self, obj):
...             return True
...
>>> isinstance([], G())
True
```

It seems RustPython isn't checking for the presence of the method before raising the exception:

```python
>>>>> class G:
.....     def __instancecheck__(self, obj):
.....             return True
.....
>>>>> isinstance([], G())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: isinstance() arg 2 must be a type or tuple of types, not G
```

In the typing module `_GenericAlias` isn't a subclass of `type` but has a ``__instancecheck__`` method.

## Python Documentation

The docs are a bit unclear, but relevant parts are:

* https://docs.python.org/3/library/functions.html#isinstance
* https://docs.python.org/3/reference/datamodel.html#class.__instancecheck__
* https://docs.python.org/3/glossary.html#term-abstract-base-class

and [PEP 3119](https://www.python.org/dev/peps/pep-3119/)

**Comment by Snowapril** [comment]
Hi. I fixed this issue in #2952 . Could you check it one more time? 😊

**Comment by domdfcoding** [comment]
Yep, seems to have sorted it. Thanks
